modules/exploits/windows/http/exchange_chainedserializationbinder_blacklist_typo_rce.rb

```diff
@@ -0,0 +1,208 @@
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/owa/auth/logon.aspx')
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    if res.code == 200 && ((%r{/owa/(?<build>\d+\.\d+\.\d+\.\d+)} =~ res.body) || (%r{/owa/auth/(?<build>\d+\.\d+\.\d+\.\d+)} =~ res.body))
+      if vuln_builds.any? { |build_range| Rex::Version.new(build).between?(*build_range) }
+        return CheckCode::Appears("Exchange Server #{build} is a vulnerable build.")
+      else
+        return CheckCode::Safe("Exchange Server #{build} is not a vulnerable build.")
+      end
+    end
+
+    # Next try @tseller's way and try /ecp/Current/exporttool/microsoft.exchange.ediscovery.exporttool.application
+    # URL which if successful should provide some XML with entries like the following:
+    #
+    # <assemblyIdentity name="microsoft.exchange.ediscovery.exporttool.application"
+    # version="15.2.986.5" publicKeyToken="b1d1a6c45aa418ce" language="neutral"
+    # processorArchitecture="msil" xmlns="urn:schemas-microsoft-com:asm.v1" />
+    #
+    # This only works on Exchange Server 2013 and later and may not always work, but if it
+    # does work it provides the full version number so its a nice strategy.
+    #
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/ecp/current/exporttool/microsoft.exchange.ediscovery.exporttool.application')
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    if res.code == 200 && res.body =~ /name="microsoft.exchange.ediscovery.exporttool" version="\d+\.\d+\.\d+\.\d+"/
+      build = res.body.match(/name="microsoft.exchange.ediscovery.exporttool" version="(\d+\.\d+\.\d+\.\d+)"/)[1]
+      if vuln_builds.any? { |build_range| Rex::Version.new(build).between?(*build_range) }
+        return CheckCode::Appears("Exchange Server #{build} is a vulnerable build.")
+      else
+        return CheckCode::Safe("Exchange Server #{build} is not a vulnerable build.")
+      end
+    end
+
+    # Finally, try a variation on the above and use a well known trick of grabbing /owa/auth/logon.aspx
+    # to get a partial version number, then use the URL at /ecp/<version here>/exporttool/. If we get a 200
+    # OK response, we found the target version number, otherwise we didn't find it.
+
+    for build in ['15.1.2308.8', '15.1.2308.14', '15.1.2308.15', '15.1.2375.7', '15.1.2375.12', '15.2.922.7', '15.2.922.13', '15.2.922.14', '15.2.986.5', '15.2.986.9']
+      res = send_request_cgi(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, "/ecp/#{build}/exporttool/")
+      )
+
+      unless res
+        return CheckCode::Unknown('Target did not respond to check.')
+      end
+
+      if res && res.code == 200
+        return CheckCode::Appears("Exchange Server #{build} is a vulnerable build.")
+      end
+    end
```

Comment:
```suggestion
   vuln_builds.each do |version_range|
      prepend_range = version_range[0].canonical_segments[..-2]
      lowest_patch = version_range[0].canonical_segments.last
      while Rex::Version.new((prepend_range.dup << lowest_patch).join('.')) <= version_range[1]
        res = send_request_cgi(
          'method' => 'GET',
          'uri' => normalize_uri(target_uri.path, "/ecp/#{build}/exporttool/")
        )

        unless res
          return CheckCode::Unknown('Target did not respond to check.')
        end

        if res && res.code == 200
          return CheckCode::Appears("Exchange Server #{build} is a vulnerable build.")
        end
        lowest_patch += 1
      end
    end
```

Reply:
Implementation added in https://github.com/rapid7/metasploit-framework/pull/16164/commits/4c1b2478fa50bb6f0a6bd82ecd02acff5094d094, thanks @jmartin-r7 for the advice on this one!